ooyala_player/ooyala_player.py

```diff
@@ -358,15 +353,38 @@ class OoyalaPlayerBlock(OoyalaPlayerMixin, XBlock):
     xml_config = String(help="XML Configuration", default='<ooyala>\n</ooyala>',
                         scope=Scope.content)
 
+    def _get_attr_or_default(self, attribute_name, setting_name):
+        """
+        If the given attribute is set on the instance, return it immediately.
+        Otherwise check if it has a default in the settings service.
+        If neither are available, return None
+        """
+        available_attr = getattr(self, attribute_name)
+        if available_attr:
+            return available_attr
+        settings_service = self.runtime.service(self, 'settings')
+        if settings_service:
+            return settings_service.get_settings_bucket(self).get(setting_name)
+
     @property
     def api_key_3play_with_default_setting(self):
-        if self.api_key_3play:
-            return self.api_key_3play
+        return self._get_attr_or_default('api_key_3play', '3PLAY_API_KEY')
 
-        settings_service = self.runtime.service(self, 'settings')
-        if settings_service:
-            return settings_service.get_settings_bucket(self).get('3PLAY_API_KEY')
-        return None
+    @property
+    def enable_player_token_with_default_setting(self):
+        return self._get_attr_or_default('enable_player_token', 'PLAYER_TOKEN')
```

Comment:
Why `PLAYER_TOKEN` instead of `ENABLE_PLAYER_TOKEN`? I think `ENABLE_PLAYER_TOKEN` makes it more obvious that the value is a boolean.